UPGRADE-API-1.12.md

```diff
@@ -0,0 +1,8 @@
+# UPGRADE FROM `v1.11.X` TO `v1.12.0`
+
+1. Service `src/Sylius/Bundle/ApiBundle/DataProvider/CartPaymentMethodsSubresourceDataProvider.php` has been removed and logic was replaced by `src/Sylius/Bundle/ApiBundle/DataProvider/PaymentMethodsCollectionDataProvider.php`
+
+2. Endpoints `/shop/orders/{tokenValue}/payments/{payments}/methods`, `/shop/payments/{id}/methods` has been removed and replaced by `/shop/payment-methods` with filter `paymentId` and `tokenValue`
```

Comment:
Make this upgrade note consistent with changes in shipping methods after rebase :)